sync/class.jetpack-sync-defaults.php

```diff
@@ -128,9 +133,22 @@ class Jetpack_Sync_Defaults {
 		'DISABLE_WP_CRON',
 		'ALTERNATE_WP_CRON',
 		'WP_CRON_LOCK_TIMEOUT',
-		'PHP_VERSION'
+		'PHP_VERSION',
 	);
 
+	public static function get_constants_whitelist() {
+		/**
+		 * Filter the list of PHP constants that are manageable via the JSON API.
+		 *
+		 * @module sync
+		 *
+		 * @since 4.7
+		 *
+		 * @param array The default list of constants options.
+		 */
+		return apply_filters( 'jetpack_constants_whitelist', self::$default_constants_whitelist );
```

Comment:
Should we maybe name the filter `jetpack_sync_constants` or the like?  Whitelist doesn't seem hugely relevant -- as that's the nature of having a list we're filtering -- but including `sync` in the filter could probably help with clarity.